tests/test_datasets.py

```diff
@@ -359,3 +364,179 @@ def test_dpo_loader_pad_to_multiple(init_model_parallel, make_tmp_jsonl, str_to_
         num_mini_batches += 1
 
     assert num_mini_batches == 2
+
+
+@pytest.mark.run_only_on("GPU")
+def test_packed_dpo_loader(init_model_parallel, make_tmp_jsonl, llama3_tokenizer):
```

Comment:
i guess you don't use `make_tmp_jsonl` so you can remove it, but actually below i see you create a tmp directory. any reason not to just use the [tmp_path](https://docs.pytest.org/en/stable/how-to/tmp_path.html) fixture?

Reply:
Good point. I updated the tests to use `tmp_path`.